src/runtime/virtcontainers/mount.go

```diff
@@ -194,24 +194,40 @@ func getDeviceForPath(path string) (device, error) {
 	return dev, nil
 }
 
-var blockFormatTemplate = "/sys/dev/block/%d:%d/dm"
+var checkStorageDriver = checkBlockDevice
 
-var checkStorageDriver = isDeviceMapper
-
-// isDeviceMapper checks if the device with the major and minor numbers is a devicemapper block device
-func isDeviceMapper(major, minor int) (bool, error) {
+var blockFormatTemplates = map[string]string{
+	"dm":    "/sys/dev/block/%d:%d/dm",
+	"block": "/sys/dev/block/%d:%d/device/block",
+	"loop":  "/sys/dev/block/%d:%d/loop",
+}
 
-	//Check if /sys/dev/block/${major}-${minor}/dm exists
-	sysPath := fmt.Sprintf(blockFormatTemplate, major, minor)
+func isBlockDevice(major, minor int, formatTemplate string) (bool, error) {
+	sysPath := fmt.Sprintf(formatTemplate, major, minor)
```

Comment:
It does not seem to work for nvme devices. Why not just check for `/sys/dev/block/<major:minor>/` and forget about the specific block device types?

Reply:
I have actually tested the above 3 block types with kata, and I'm not sure kata is compatible with all block types.
Just to be cautious, I make a minimal supportation, and allow for extensions.

Reply:
All block devices are equal in terms of virtio-block backend. So it doesn't matter where it comes from.

Reply:
OK, it seems reasonable.